ddl/db_test.go

```diff
@@ -1850,6 +1850,60 @@ func (s *testDBSuite4) TestChangeColumn(c *C) {
 	s.tk.MustExec("drop table t3")
 }
 
+func (s *testDBSuite5) TestRenameColumn(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use " + s.schemaName)
+
+	assertColNames := func(tableName string, colNames ...string) {
+		cols := s.testGetTable(c, tableName).Cols()
+		c.Assert(len(cols), Equals, len(colNames), Commentf("number of columns mismatch"))
+		for i := range cols {
+			c.Assert(cols[i].Name.L, Equals, strings.ToLower(colNames[i]))
+		}
+	}
+
+	s.mustExec(c, "create table test_rename_column (id int not null primary key auto_increment, col1 int)")
+	s.mustExec(c, "alter table test_rename_column rename column col1 to col1")
+	assertColNames("test_rename_column", "id", "col1")
+	s.mustExec(c, "alter table test_rename_column rename column col1 to col2")
+	assertColNames("test_rename_column", "id", "col2")
+
+	// Test renaming non-exist columns.
+	s.tk.MustGetErrCode("alter table test_rename_column rename column non_exist_col to col3", mysql.ErrBadField)
+
+	// Test renaming to an exist column.
+	s.tk.MustGetErrCode("alter table test_rename_column rename column col2 to id", mysql.ErrDupFieldName)
+
+	// Test renaming the column with foreign key.
+	s.tk.MustExec("drop table test_rename_column")
+	s.tk.MustExec("create table test_rename_column_base (base int)")
+	s.tk.MustExec("create table test_rename_column (col int, foreign key (col) references test_rename_column_base(base))")
+
+	s.tk.MustGetErrCode("alter table test_rename_column rename column col to col1", mysql.ErrFKIncompatibleColumns)
+
+	s.tk.MustExec("drop table test_rename_column_base")
+
+	// Test renaming generated columns.
+	s.tk.MustExec("drop table test_rename_column")
+	s.tk.MustExec("create table test_rename_column (id int, col1 int generated always as (id + 1))")
```

Comment:
I mean, rename `id`, not `col1`, because `col1` refers to `id`.

Reply:
Done. PTAL